igvc_navigation/src/mapper/main.cpp

```diff
@@ -54,8 +57,8 @@ void odom_callback(const nav_msgs::Odometry::ConstPtr &msg)
 {
   if (update)
```

Comment:
I'm a bit confused about what you mean here. Are you saying that I should be synchronizing the timestamps from the odom message and the point cloud message? That would require a much larger change to the logic of the file right?

Reply:
Yes this will be a significant change and we will have to validate it to make sure it works like I think it does.

You should be using the position message that most closely matches the timestamp of the sensor message (i.e. lidar or camera). Fortunately TF will handle this for you. What you would do here is get the transform from sensor to /odom each time the callback is called and then use that transform to get the points in to the global reference frame. TF will do linear interpolation between the two most recent transforms relative to the time you are requesting it at. Which should be close enough in theory.

Reply:
Realistically we should implement the TF method and compare that to the current method and see which gives less error

Reply:
This would be done using `sync_policies` right? Do you if exact time or approximate time works better here?

Reply:
I'm actually a bit confused about where `odometry/filtered` is being published from, since I can't really find it. What's sending messages in that topic

Reply:
EKF localization, a node we include and run (localization.launch) but we did not write ourselves. It is just the best guess of the current position.

Reply:
I'm still quite confused about what I need to do. My guess is that I do `lookupTransform` on the transform published by the localization node (not sure what the name of this is), and then I use the results of that transform rather than the messages I am getting from `/odometry/filtered`?